### PR TITLE
Add `--no-build` to dotnet test command

### PR DESCRIPTION
**Pull request by emanuelgaspar**
If this works, I owe @TiagoFroisPereira a drink.

### CODE REVIEW

File: test-dotnet/action.yml [hunk 1]
```diff
@@ -70,7 +70,7 @@ runs:
         for f in $(find . -name "*.Tests.csproj"); do echo "testing project $f" && \
           csprojName=${f##*/} && \
           projectName="${csprojName/.csproj/}" && \
-          dotnet test $f --configuration Debug --no-restore --logger GitHubActions --collect "XPlat Code Coverage" --results-directory "TestResults/$projectName"
+          dotnet test $f --configuration Debug --no-build --logger GitHubActions --collect "XPlat Code Coverage" --results-directory "TestResults/$projectName"
```
Comment (TiagoFroisPereira) [inline test-dotnet/action.yml:73]: Need to be both the --no-restore and --no-build

Comment (emanuelgaspar) [reply test-dotnet/action.yml:73]: Not according to the docs:

**--no-build**
Doesn't build the project before publishing. **It also implicitly sets the --no-restore flag.**

<https://learn.microsoft.com/en-us/dotnet/core/tools/dotnet-publish#:~:text=for%20each%20manifest.-,%2D%2Dno%2Dbuild,-Doesn%27t%20build%20the>

**Review by TiagoFroisPereira — APPROVED** [review]
Let's go

**Comment by emanuelgaspar** [comment]
Demoed here:
<https://github.com/trakx/market-data/actions/runs/7584705266/job/20658978783#step:3:329>

The test took less than 5 minutes, so much better than 10+ minutes before 😃

**Pull request merged**